### DISCUSSION TITLE
Should V8.* work with WPF and .NET Framework 4.7.1

### DISCUSSION THREAD

**Discussion by JPhilC** [CommunityToolkit/dotnet#500] [Q&A]
Hi all,

Firstly, I am using Visual Studio 2022 (17.4.0)

I have a product that has been in production for some time. The solution consists of a WPF Desktop app and several class libraries which all target .NET Framework 4.7.1. Some of the referenced class libraries are also used by libraries that provide add-ons for AutoCAD and MicroStation so, for the time being, I am constrained to using .NET Framework 4.7.1.

Originally the application used MVVMLight but a while back I converted to Microsoft.Toolkit.Mvvm and I have just recently migrated to Community.Toolkit.Mvvm. 

I had to roll back to V7.1.2 though to get a clean build after updating all my namespace references. Version 8.0.0 would not build cleanly. Initially the build complained because I was not using C#8 or later. V8.1.0-preview1 fixed this but then I started getting numerous duplicate NotNullAttribute errors. At this point I just rolled back as I do not have time at the moment to dig any further.

Simple question, should I be able to use V8,* with .NET Framework 7.4.1 or am I on a hiding to nothing and should stick at V7.1.2?

**Comment by michael-hawker** [comment]
Sorry for the delayed response.

@JPhilC if you're using package reference, you should be fine. You probably just have to update the `LangVersion` in your csproj file?
